### PR TITLE
don't load environment of children for custom anchor function

### LINKED ISSUE #1622: Custom anchor function on a group/dynamicgroup break weakauras initialization
**Describe the bug**
I apologize but I don't know the exact cause. I'm guessing there was a silent error somewhere. I was messing with Custom Frame, Custom Anchor, and Custom Grow settings. I was AFK when I disconnected and don't remember exactly what I had enabled.

**Do you have an error log of what happened?**

```Message: Interface\AddOns\WeakAuras\AuraEnvironment.lua:189: attempt to index field '?' (a nil value)
Interface\AddOns\WeakAuras\AuraEnvironment.lua:189: in function `ActivateAuraEnvironment'
Interface\AddOns\WeakAuras\AuraEnvironment.lua:177: in function `ActivateAuraEnvironment'
Interface\AddOns\WeakAuras\WeakAuras.lua:6436: in function <Interface\AddOns\WeakAuras\WeakAuras.lua:6387>
Interface\AddOns\WeakAuras\WeakAuras.lua:6454: in function `AnchorFrame'
...ace\AddOns\WeakAuras\RegionTypes\RegionPrototype.lua:430: in function `modify'
...erface\AddOns\WeakAuras\RegionTypes\DynamicGroup.lua:686: in function `modify'
Interface\AddOns\WeakAuras\WeakAuras.lua:4129: in function `SetRegion'
Interface\AddOns\WeakAuras\WeakAuras.lua:3978: in function <Interface\AddOns\WeakAuras\WeakAuras.lua:3954>
Interface\AddOns\WeakAuras\WeakAuras.lua:4060: in function `Add'
Interface\AddOns\WeakAuras\WeakAuras.lua:3614: in function <Interface\AddOns\WeakAuras\WeakAuras.lua:3593>
Interface\AddOns\WeakAuras\WeakAuras.lua:3606: in function `load'
Interface\AddOns\WeakAuras\WeakAuras.lua:3621: in function `AddMany'
Interface\AddOns\WeakAuras\WeakAuras.lua:1521: in function <Interface\AddOns\WeakAuras\WeakAuras.lua:1509>

Time: Mon Aug 26 14:08:22 2019
Count: 1
Stack: Interface\AddOns\WeakAuras\AuraEnvironment.lua:189: attempt to index field '?' (a nil value)
Interface\AddOns\WeakAuras\AuraEnvironment.lua:189: in function `ActivateAuraEnvironment'
Interface\AddOns\WeakAuras\AuraEnvironment.lua:177: in function `ActivateAuraEnvironment'
Interface\AddOns\WeakAuras\WeakAuras.lua:6436: in function <Interface\AddOns\WeakAuras\WeakAuras.lua:6387>
Interface\AddOns\WeakAuras\WeakAuras.lua:6454: in function `AnchorFrame'
...ace\AddOns\WeakAuras\RegionTypes\RegionPrototype.lua:430: in function `modify'
...erface\AddOns\WeakAuras\RegionTypes\DynamicGroup.lua:686: in function `modify'
Interface\AddOns\WeakAuras\WeakAuras.lua:4129: in function `SetRegion'
Interface\AddOns\WeakAuras\WeakAuras.lua:3978: in function <Interface\AddOns\WeakAuras\WeakAuras.lua:3954>
Interface\AddOns\WeakAuras\WeakAuras.lua:4060: in function `Add'
Interface\AddOns\WeakAuras\WeakAuras.lua:3614: in function <Interface\AddOns\WeakAuras\WeakAuras.lua:3593>
Interface\AddOns\WeakAuras\WeakAuras.lua:3606: in function `load'
Interface\AddOns\WeakAuras\WeakAuras.lua:3621: in function `AddMany'
Interface\AddOns\WeakAuras\WeakAuras.lua:1521: in function <Interface\AddOns\WeakAuras\WeakAuras.lua:1509>
```

**To Reproduce**
 
SV is provided.

**Screenshots**

**Did you try having WeakAuras as the only enabled addon and everything else (especially something like ElvUI) disabled?**

Tested only with the 5 official WAs enabled.

**Which version of WeakAuras are you using?**

2.14.4

**Was it working in a previous version? If yes, which was the last good one?**

**Additional context**

[WeakAurasSV.zip](https://github.com/WeakAuras/WeakAuras2/files/3542759/WeakAurasSV.zip)

### PR DESCRIPTION
**Pull request by mrbuds**
fix #1622

Custom anchor function is run on group's Add, before children are added
It activate environment for the group which activate environment for it's children, but that don't work because they are not added yet.

This skip activating environment for group children for this custom function.

### CODE REVIEW

**Comment by InfusOnWoW** [comment]
I'm not sure what the right solution is. A few random thoughts:
* how does this works on importing?
* technically the AnchorFrame custom function doesn't really need the child aura envs
* the other custom functions in dynamic group are paused and we run ReloadControlledChildren at the end. (iirc)

**Comment by InfusOnWoW** [comment]
This doesn't work, as the next time ActivateAuraEnvironment is called, environment_initialized will be true, and thus the children's environemts won't be added.

File: WeakAuras/WeakAuras.lua [hunk 1]
```diff
@@ -6432,12 +6432,12 @@ local function GetAnchorFrame(region, anchorFrameType, parent, anchorFrameFrame)
 
   if (anchorFrameType == "CUSTOM" and region.customAnchorFunc) then
     WeakAuras.StartProfileSystem("custom region anchor")
```
Comment (emptyrivers) [inline WeakAuras/WeakAuras.lua]: I was expecting something like this, honestly.
```suggestion
	if not WeakAuras.IsLoginFinished() then
		return parent
	else
	    WeakAuras.StartProfileSystem("custom region anchor")
	    -- etc
```

Comment (mrbuds) [reply WeakAuras/WeakAuras.lua]: i want to implement this, but i'm leveling my priest :)

**Comment by mrbuds** [comment]
I reverted (not deleted) first 2 commits, it needs a squash before merge

**Pull request merged**